model/parts/update/update_graph.go

```diff
@@ -29,19 +29,12 @@ func Graph(state *types.State, requestResult types.RequestResult, curTimeStep in
 					continue
 				} else {
 					if !config.IsPayOnlyForCurrentRequest() {
-						newEdgeData1 := edgeData1
-						newEdgeData1.A2B = 0
-						state.Graph.SetEdgeData(payment.FirstNodeId, payment.PayNextId, newEdgeData1)
-
-						newEdgeData2 := edgeData2
-						newEdgeData2.A2B = 0
-						state.Graph.SetEdgeData(payment.PayNextId, payment.FirstNodeId, newEdgeData2)
+						state.Graph.SetEdgeA2B(payment.FirstNodeId, payment.PayNextId, 0)
+						state.Graph.SetEdgeA2B(payment.PayNextId, payment.FirstNodeId, 0)
 					} else {
 						// Important fix: Reduce debt here, since it debt will be added again below.
 						// Idea is, paying for the current request should not effect the edge balance.
-						newEdgeData1 := edgeData1
-						newEdgeData1.A2B = edgeData1.A2B - price
-						state.Graph.SetEdgeData(payment.FirstNodeId, payment.PayNextId, newEdgeData1)
```

Comment:
Here, you have removed some code while Arman had them in the latest version he pushed. Should we keep your code?